Add plugin: Insta TOC

iLiftALot opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/iLiftALot/insta-toc

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[const mainManifest = manifest ?? Process.env.pluginManifest;](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/main.ts#L27)
What is this for?
It makes no sense to me to have this.

[.setName('List Bullet Type')](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/SettingsTab.ts#L26), [.setName('Indent Size')](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/SettingsTab.ts#L39), [.setName('Update Delay')](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/SettingsTab.ts#L54)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

iLiftALot commented:
> [const mainManifest = manifest ?? Process.env.pluginManifest;](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/main.ts#L27) What is this for? It makes no sense to me to have this.
> 
> [.setName('List Bullet Type')](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/SettingsTab.ts#L26), [.setName('Indent Size')](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/SettingsTab.ts#L39), [.setName('Update Delay')](https://github.com/iLiftALot/insta-toc/blob/c878193933db693aa7f3d1c2d13e5ebc045f58c5/src/SettingsTab.ts#L54) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Got it. I've removed the `Process` object and adjusted the naming conventions for the settings.

That custom `Process` object is just something I sometimes use universally across environments as a workaround for accessing global .env variables on mobile. However, it's unnecessary for the circumstances of this plugin.

Let me know if the setting names need more work.
